File: pentalk_front/android/app/src/main/kotlin/com/example/pentalk_front/drawing/DrawingChannel.kt
package com.example.pentalk_front.drawing

import io.flutter.plugin.common.MethodChannel

object DrawingChannel {
    const val CHANNEL_NAME = "pentalk/native_drawing"

    @Volatile
    var channel: MethodChannel? = null

    fun notifyToolChanged(
        tool: String,
        source: String,
        color: Int? = null,
        size: Float? = null,
        eraserSize: Float? = null,
    ) {
        val payload = mutableMapOf<String, Any>(
            "tool" to tool,
            "source" to source,
        )
        if (color != null) payload["color"] = color
        if (size != null) payload["size"] = size
        if (eraserSize != null) payload["eraserSize"] = eraserSize
        channel?.invokeMethod("toolChanged", payload)
    }
}

File: pentalk_front/ios/Runner/DrawingChannel.swift
import Flutter

enum DrawingChannel {
    static let name = "pentalk/drawing"
    static var channel: FlutterMethodChannel?

    static func notifyToolChanged(
        tool: String,
        source: String,
        color: Int? = nil,
        size: CGFloat? = nil,
        eraserSize: CGFloat? = nil
    ) {
        var payload: [String: Any] = [
            "tool": tool,
            "source": source,
        ]
        if let color = color { payload["color"] = color }
        if let size = size { payload["size"] = size }
        if let eraserSize = eraserSize { payload["eraserSize"] = eraserSize }
        channel?.invokeMethod("toolChanged", arguments: payload)
    }

    static func notifyDrawEvent(_ payload: [String: Any]) {
        channel?.invokeMethod("onDrawEvent", arguments: payload)
    }
}
